src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs

```diff
@@ -163,6 +163,11 @@ public static FormulaValue DateValue_UO(IRContext irContext, UntypedObjectValue[
 
                 if (IsValidDateTimeUO(s) && DateTime.TryParse(s, CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime res))
                 {
+                    if (!res.IsValid(runner))
+                    {
+                        return CommonErrors.InvalidDateTimeError(irContext);
+                    }
+
```

Comment:
DateParse on LibraryDate.cs doesn't validate the date - this should follow the same semantics of it